site/pages/admin/social-media-accounts.page.tsx

```diff
@@ -1,51 +1,180 @@
 
 const title = "Social Media Accounts - Create Transport Disruptions Service";
 const description = "Social Media Accounts page for the Create Transport Disruptions Service";
 
+export const getHootsuiteData = async (
+    ctx: NextPageContext,
+    username: string,
+    orgId: string,
+): Promise<{ clientId: string; userData: SocialMediaAccountsSchema }> => {
+    const cookies = parseCookies(ctx);
+    let clientIdValue = "";
+    let userData: SocialMediaAccountsSchema = [];
+    try {
+        const idToken = cookies[COOKIES_ID_TOKEN];
+        const refreshToken = cookies[COOKIES_REFRESH_TOKEN];
+
+        const clientId = await getParameter(`/social/hootsuite/client_id`);
+        const clientSecret = await getParameter(`/social/hootsuite/client_secret`);
+        if (!clientId || !clientSecret) {
+            throw new Error("clientId and clientSecret must be defined");
+        }
+
+        clientIdValue = clientId.Parameter?.Value || "";
+        const key = `${clientId.Parameter?.Value || ""}:${clientSecret.Parameter?.Value || ""}`;
+
+        const authToken = `Basic ${Buffer.from(key).toString("base64")}`;
+
+        if (idToken && refreshToken)
+            await Promise.all([
+                putParameter(`/${username}/token`, idToken, "SecureString", true),
+                putParameter(`/${username}/refresh-token`, refreshToken, "SecureString", true),
+            ]);
+
+        const tokensByOrganisation = await getParametersByPath(`/social/${orgId}/hootsuite`);
+        const refreshTokens = tokensByOrganisation?.Parameters?.map((token) => {
+            return {
+                value: token.Value,
+                name: token.Name,
+                userId: token?.Name?.split("hootsuite/")[1]?.split("-")[0] ?? "",
+                accountType: startCase(token?.Name?.split("/")[3]) ?? "",
+                addedBy: token?.Name?.split("/")[4]?.replace("_", " ")?.split("-")[1] ?? "",
+            };
+        });
+
+        if (refreshTokens && refreshTokens.length > 0) {
+            await Promise.all(
+                refreshTokens?.map(async (token) => {
+                    const resp = await fetch(`${HOOTSUITE_URL}oauth2/token`, {
+                        method: "POST",
+                        body: new URLSearchParams({
+                            grant_type: "refresh_token",
+                            refresh_token: token.value ?? "",
+                        }),
+                        headers: {
+                            "Content-Type": "application/x-www-form-urlencoded",
+                            Authorization: authToken,
+                        },
+                    });
+                    if (resp.ok) {
+                        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+                        const tokenResult: HootsuiteToken = await resp.json();
+                        const keys = await getParametersByPath(`/social/${orgId}/hootsuite`);
```

Comment:
This api call can be included in the promise.all above

Reply:
done